sbt-plugin/src/sbt-test/gen-java/02-server-reflection/src/main/java/example/myapp/helloworld/Main.java

```diff
@@ -0,0 +1,54 @@
+package example.myapp.helloworld;
+
+import java.util.concurrent.CompletionStage;
+
+import akka.actor.ActorSystem;
+
+import akka.stream.ActorMaterializer;
+import akka.stream.Materializer;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+//#server-reflection
+import java.util.Arrays;
+
+import akka.grpc.javadsl.ServiceHandler;
+import akka.grpc.javadsl.ServerReflection;
+import akka.http.javadsl.*;
+
+import example.myapp.helloworld.grpc.*;
+
+//#server-reflection
+
+public class Main {
+    public static void main(String[] args) throws Exception {
+        // important to enable HTTP/2 in ActorSystem's config
+        Config conf = ConfigFactory.parseString("akka.http.server.preview.enable-http2 = on")
+            .withFallback(ConfigFactory.defaultApplication());
+        // Akka ActorSystem Boot
+        ActorSystem sys = ActorSystem.create("HelloWorld", conf);
+
+        run(sys).thenAccept(binding -> {
+            System.out.println("gRPC server bound to: " + binding.localAddress());
+        });
+    }
+
+    public static CompletionStage<ServerBinding> run(ActorSystem sys) throws Exception {
+        Materializer mat = ActorMaterializer.create(sys);
+
+        //#server-reflection
+        // Instantiate implementation
+        GreeterService impl = new GreeterServiceImpl();
+
+        // Bind service handler servers to localhost:8080
+        return Http.get(sys).bindAndHandleAsync(
+            ServiceHandler.concatOrNotFound(
+                GreeterServiceHandlerFactory.create(impl, mat, sys),
+                new ServerReflection(java.util.Arrays.asList(GreeterService.service)).create(mat, sys)
+            ),
```

Comment:
Good showing the server reflection second as that will likely be hit less often than the actual service

Reply:
yes - and where it's hit it's probably not on a performance-critical path